pkg/apis/controller/experiments/v1beta1/experiment_types.go

```diff
@@ -187,8 +188,36 @@ type FeasibleSpace struct {
 }
 
 type TrialTemplate struct {
-	Retain     bool        `json:"retain,omitempty"`
+	// Retain indicates that Trial resources must be not cleanup
+	Retain bool `json:"retain,omitempty"`
+
 	GoTemplate *GoTemplate `json:"goTemplate,omitempty"`
+
+	// List of parameres that are used in Trial template
+	TrialParameters []TrialParameterSpec `json:"trialParameters,omitempty"`
+
+	// Trial spec contains Trial template in unstructured format
+	TrialSpec *unstructured.Unstructured `json:"trialSpec,omitempty"`
+
+	// Name of config map where Trial template is located
+	ConfigMapName string `json:"configMapName,omitempty"`
```

Comment:
The field is already in GoTemplate, should we remove it here?

Reply:
Yeah, but I think when we implement new logic we will not support `GoTemplate` ?

Reply:
SGTM, then can we just remove gotemplate?